From 9146e7016689e1a04345640ebc25068e9674ff00 Mon Sep 17 00:00:00 2001
From: Michael Larabel <michael@phoronix.com>
Date: Sat, 18 Feb 2023 05:34:49 -0600
Subject: [PATCH] Cleanup
 https://github.com/phoronix-test-suite/phoronix-test-suite/pull/699

The install on macOS having AppStream file was superfluous and no point in installing the XML to the modern "metainfo" location if the existing AppStream data wasn't even valid against their modern standards

Closes: https://github.com/phoronix-test-suite/phoronix-test-suite/pull/699
---
 deploy/rpm-package/build-package-rpm.php      |  2 +-
 install-sh                                    |  4 +--
 install_macos                                 |  2 --
 ...est_suite.phoronix_test_suite.metainfo.xml | 28 +++++++++++++++++++
 .../static/phoronix-test-suite.appdata.xml    | 20 -------------
 5 files changed, 31 insertions(+), 25 deletions(-)
 create mode 100644 pts-core/static/com.phoronix_test_suite.phoronix_test_suite.metainfo.xml
 delete mode 100644 pts-core/static/phoronix-test-suite.appdata.xml

diff --git a/deploy/rpm-package/build-package-rpm.php b/deploy/rpm-package/build-package-rpm.php
index df779a63b0..454786ec8f 100644
--- a/deploy/rpm-package/build-package-rpm.php
+++ b/deploy/rpm-package/build-package-rpm.php
@@ -75,7 +75,7 @@
 $spec_file .= "%{_datadir}/phoronix-test-suite/*\n";
 $spec_file .= "%{_datadir}/applications/*\n";
 $spec_file .= "%{_datadir}/icons/hicolor/*\n";
-$spec_file .= "%{_datadir}/appdata/%{name}.appdata.xml\n";
+$spec_file .= "%{_datadir}/metainfo/com.phoronix_test_suite.phoronix_test_suite.metainfo.xml\n";
 $spec_file .= "%{_datadir}/doc/*\n";
 $spec_file .= "%{_mandir}/man1/%{name}.1*\n";
 $spec_file .= "%config(noreplace) %{_sysconfdir}/bash_completion.d\n";
diff --git a/install-sh b/install-sh
index f6228cde94..caf8e08b72 100755
--- a/install-sh
+++ b/install-sh
@@ -50,7 +50,7 @@ rm -rf $DESTDIR$INSTALL_PREFIX/share/phoronix-test-suite/
 rm -rf $DESTDIR$INSTALL_PREFIX/share/doc/phoronix-test-suite/
 
 mkdir -p $DESTDIR$INSTALL_PREFIX/bin/
-mkdir -p $DESTDIR$INSTALL_PREFIX/share/appdata/
+mkdir -p $DESTDIR$INSTALL_PREFIX/share/metainfo/
 mkdir -p $DESTDIR$INSTALL_PREFIX/share/applications/
 mkdir -p $DESTDIR$INSTALL_PREFIX/share/icons/hicolor/48x48/apps/
 mkdir -p $DESTDIR$INSTALL_PREFIX/share/man/man1/
@@ -74,7 +74,7 @@ cp pts-core/static/bash_completion $DESTDIR$INSTALL_PREFIX/../etc/bash_completio
 cp pts-core/static/images/phoronix-test-suite.png $DESTDIR$INSTALL_PREFIX/share/icons/hicolor/48x48/apps/phoronix-test-suite.png
 cp pts-core/static/phoronix-test-suite.desktop $DESTDIR$INSTALL_PREFIX/share/applications/
 cp pts-core/static/phoronix-test-suite-launcher.desktop $DESTDIR$INSTALL_PREFIX/share/applications/
-cp pts-core/static/phoronix-test-suite.appdata.xml $DESTDIR$INSTALL_PREFIX/share/appdata/
+cp pts-core/static/com.phoronix_test_suite.phoronix_test_suite.metainfo.xml $DESTDIR$INSTALL_PREFIX/share/metainfo/
 
 mkdir -p $DESTDIR$INSTALL_PREFIX/../usr/lib/systemd/system/
 cp deploy/*-systemd/*.service $DESTDIR$INSTALL_PREFIX/../usr/lib/systemd/system/
diff --git a/install_macos b/install_macos
index 62dd9e12e5..8ec4147cae 100755
--- a/install_macos
+++ b/install_macos
@@ -51,7 +51,6 @@ rm -rf "$DESTDIR$INSTALL_PREFIX"/share/phoronix-test-suite/
 rm -rf "$DESTDIR$INSTALL_PREFIX"/share/doc/phoronix-test-suite/
 
 mkdir -p "$DESTDIR$INSTALL_PREFIX"/bin/
-mkdir -p "$DESTDIR$INSTALL_PREFIX"/share/appdata/
 mkdir -p "$DESTDIR$INSTALL_PREFIX"/share/applications/
 mkdir -p "$DESTDIR$INSTALL_PREFIX"/share/icons/hicolor/48x48/apps/
 mkdir -p "$DESTDIR$INSTALL_PREFIX"/share/man/man1/
@@ -75,7 +74,6 @@ cp pts-core/static/bash_completion "$DESTDIR$INSTALL_PREFIX"/../etc/bash_complet
 cp pts-core/static/images/phoronix-test-suite.png "$DESTDIR$INSTALL_PREFIX"/share/icons/hicolor/48x48/apps/phoronix-test-suite.png
 cp pts-core/static/phoronix-test-suite.desktop "$DESTDIR$INSTALL_PREFIX"/share/applications/
 cp pts-core/static/phoronix-test-suite-launcher.desktop "$DESTDIR$INSTALL_PREFIX"/share/applications/
-cp pts-core/static/phoronix-test-suite.appdata.xml "$DESTDIR$INSTALL_PREFIX"/share/appdata/
 
 mkdir -p "$DESTDIR$INSTALL_PREFIX"/../usr/lib/systemd/system/
 cp deploy/*-systemd/*.service "$DESTDIR$INSTALL_PREFIX"/../usr/lib/systemd/system/
diff --git a/pts-core/static/com.phoronix_test_suite.phoronix_test_suite.metainfo.xml b/pts-core/static/com.phoronix_test_suite.phoronix_test_suite.metainfo.xml
new file mode 100644
index 0000000000..699b64d27d
--- /dev/null
+++ b/pts-core/static/com.phoronix_test_suite.phoronix_test_suite.metainfo.xml
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="console-application">
+  <id>com.phoronix_test_suite.phoronix_test_suite</id>
+  <name>Phoronix Test Suite</name>
+  <summary>The Phoronix Test Suite is an open-source, cross-platform automated benchmarking utility</summary>
+  <metadata_license>FSFAP</metadata_license>
+  <project_license>GPL-3.0-only</project_license>
+  <content_rating>none</content_rating>
+  <description>
+    <p>
+      The Phoronix Test Suite is the most comprehensive testing and benchmarking platform available for the Linux operating system. This software is designed to effectively carry out both qualitative and quantitative benchmarks in a clean, reproducible, and easy-to-use manner.
+    </p>
+    <p>
+      The Phoronix Test Suite consists of a lightweight processing core (pts-core) with each benchmark consisting of an XML-based profile with related resource scripts. The process from the benchmark installation, to the actual benchmarking, to the parsing of important hardware and software components is heavily automated and completely repeatable, asking users only for confirmation of actions.
+    </p>
+    <p>
+      Phoronix Test Suite tests and suites are obtained via OpenBenchmarking.org and there exists support for optionally sharing your results via OpenBenchmarking.org.
+    </p>
+  </description>
+  <icon type="stock">phoronix-test-suite</icon>
+  <categories>
+    <category>System</category>
+    <category>Monitor</category>
+  </categories>
+  <provides>
+    <binary>phoronix-test-suite</binary>
+  </provides>
+</component>
diff --git a/pts-core/static/phoronix-test-suite.appdata.xml b/pts-core/static/phoronix-test-suite.appdata.xml
deleted file mode 100644
index 9d232cc7bf..0000000000
--- a/pts-core/static/phoronix-test-suite.appdata.xml
+++ /dev/null
@@ -1,20 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<!-- Copyright (C) 2008 - 2018, Michael Larabel -->
-<application>
- <id type="desktop">phoronix-test-suite.desktop</id>
- <metadata_license>FSFAP</metadata_license>
- <project_license>GPL-3.0</project_license>
- <name>Phoronix Test Suite</name>
- <summary>The Phoronix Test Suite is an open-source, cross-platform automated benchmarking utility</summary>
- <description>
-  <p>The Phoronix Test Suite is the most comprehensive testing and benchmarking platform available for the Linux operating system. This software is designed to effectively carry out both qualitative and quantitative benchmarks in a clean, reproducible, and easy-to-use manner.</p>
-  <p>The Phoronix Test Suite consists of a lightweight processing core (pts-core) with each benchmark consisting of an XML-based profile with related resource scripts. The process from the benchmark installation, to the actual benchmarking, to the parsing of important hardware and software components is heavily automated and completely repeatable, asking users only for confirmation of actions.</p>
-  <p>Phoronix Test Suite tests and suites are obtained via OpenBenchmarking.org and there exists support for optionally sharing your results via OpenBenchmarking.org.</p>
- </description>
- <screenshots>
-  <screenshot type="default" width="1400" height="788">http://www.phoronix-test-suite.com/images/phoromatic_541.png</screenshot>
- </screenshots>
- <url type="homepage">http://www.phoronix-test-suite.com/</url>
- <update_contact>michael_at_phoronix.com</update_contact>
- <translation></translation>
-</application>
